homeassistant/components/iammeter/sensor.py

```diff
@@ -0,0 +1,150 @@
+"""Support for iammeter via local API."""
+import asyncio
+
+from datetime import timedelta
+import logging
+import iammeter
+from iammeter import power_meter
+from iammeter.power_meter import IamMeterError
+import requests
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_PORT, CONF_NAME, CONF_HOST
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.event import async_track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 80
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_NAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    }
+)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Platform setup."""
+    try:
+        base = "http://admin:admin@{}:{}/monitorjson"
+        url = base.format(config[CONF_HOST], config[CONF_PORT])
+        resp = requests.get(url)
```

Comment:
This should be extracted into a separate package published on PyPi.

See: <https://developers.home-assistant.io/docs/en/creating_component_code_review.html#4-communication-with-devicesservices>

Reply:
these codes has moved to PyPi lib.

Reply:
It is not just the line, I felt like almost all content of the setup method could be extracted?